sdk/servicebus/azure-servicebus/azure/servicebus/management/_models.py

```diff
@@ -964,16 +1016,22 @@ def _from_internal_entity(cls, name, internal_rule):
         rule._internal_rule = deepcopy(internal_rule)
         return rule
 
-    def _to_internal_entity(self):
-        # type: () -> InternalRuleDescription
+    def _to_internal_entity(self, kwargs=None):
+        # type: (Optional[Dict]) -> InternalRuleDescription
+        kwargs = kwargs or {}
         if not self._internal_rule:
             self._internal_rule = InternalRuleDescription()
-        self._internal_rule.filter = self.filter._to_internal_entity() if self.filter else TRUE_FILTER  # type: ignore
+
+        rule_filter = kwargs.pop("filter", self.filter)
+        self._internal_rule.filter = rule_filter._to_internal_entity() if rule_filter else TRUE_FILTER  # type: ignore
+
+        action = kwargs.pop("action", self.action)
         self._internal_rule.action = (
-            self.action._to_internal_entity() if self.action else EMPTY_RULE_ACTION
+            action._to_internal_entity() if action else EMPTY_RULE_ACTION
         )
-        self._internal_rule.created_at = self.created_at_utc
-        self._internal_rule.name = self.name
+
+        self._internal_rule.created_at = kwargs.pop("created_at_utc", self.created_at_utc)
```

Comment:
Seems like this would be a read-only property that shouldn't be overridden from kwargs?

Reply:
the funny part is that users could still pass their own value, but the service would simply ignore it -- `created_at_utc` is unchanged.